analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt

```diff
@@ -225,11 +227,13 @@ class NuGetSupport(serviceIndexUrls: List<String> = listOf(DEFAULT_SERVICE_INDEX
                         all.details.dependencyGroups.flatMapTo(mutableSetOf()) { it.dependencies }
 
                     buildDependencyTree(
-                        referredDependencies.map {
-                            // Simply take the minimum of the Ivy-style version range.
-                            val version = it.range.removePrefix("[").substringBefore(",").trim()
+                        referredDependencies.map { dependency ->
+                            // Resolve to the highest existent version that matches the version description as specified
```

Comment:
> Resolve to the highest existent version that matches ...

The code does not know which versions do actually exist, so it cannot determine the highest existent one.

I'm highlighting that because the determined version is wrong in case an exclusive upper bound is specified for the range, e.g. `[n, m)`.

Having a hard time doing a constructive suggestion for improving it though.

Reply:
Turns out resolving to the highest version is actually wrong, and the original code was basically correct, see https://docs.microsoft.com/en-us/nuget/concepts/dependency-resolution#lowest-applicable-version.